backtrader_plotting/bokeh/figureenvelope.py

```diff
@@ -502,14 +583,49 @@ def _plot_indicator_observer(self, obj: Union[bt.Indicator, bt.Observer], master
             kwglyphs['legend_label'] = label
 
             if marker is not None:
-                kwglyphs['size'] = lineplotinfo.markersize * 1.2
-                kwglyphs['color'] = color
+                fnc_name, attrs, vals, updates = None, list(), dict(), dict()
+                if isinstance(marker, (int, float)):
+                    fnc_name, attrs, vals, updates = FigureEnvelope._mrk_fncs[int(
+                        marker)]
+                if isinstance(marker, str):
+                    fnc_name, attrs, vals, updates = FigureEnvelope._mrk_fncs[
+                        str(marker)[0]]
+                if not fnc_name or not hasattr(self.figure, fnc_name):
+                    # provide alternative methods for not available methods
+                    if fnc_name == "y":
+                        fnc_name = "text"
+                        attrs = ["text_color", "text_size"]
+                        vals.update({"text": {"value": "y"}})
+                    else:
+                        raise Exception(
+                            "Sorry, unsupported marker:"
+                            + f" '{marker}'. Please report to GitHub.")
+                        return
+                # set kwglyph values
                 kwglyphs['y'] = source_id
-
-                if marker not in FigureEnvelope._mrk_fncs:
-                    raise Exception(f"Sorry, unsupported marker: '{marker}'. Please report to GitHub.")
-                glyph_fnc_name = FigureEnvelope._mrk_fncs[marker]
-                glyph_fnc = getattr(self.figure, glyph_fnc_name)
+                for v in attrs:
+                    val = None
+                    if v in ['color', 'fill_color', 'text_color']:
+                        val = {"value": color}
+                    elif v in ['size']:
+                        val = lineplotinfo.markersize
+                    elif v in ['text_font_size']:
+                        val = {"value": "%spx" % lineplotinfo.markersize}
+                    elif v in ['text']:
+                        val = {"value": marker[1:-1]}
+                    if val is not None:
+                        kwglyphs[v] = val
+                for v in vals:
+                    val = vals[v]
+                    kwglyphs[v] = val
+                for u in updates:
+                    val = updates[u]
+                    if u in kwglyphs:
+                        kwglyphs[u] = max(
+                            1, kwglyphs[u] + val)
+                    else:
+                        kwglyphs[u] = abs(val)
```

Comment:
Hm, I am unsure about this update-mechanism... I see what it is meant for but it is somehow blurry imo. Since it is meant as an update I think the `else` clause should never happen? Maybe just throw instead of assigning the `abs` value to make sure it can really only act as an update to existing values?

Reply:
makes sense, i did not like it either. At the moment, it is just x,y and size, that gets updated. The reason why i added it, was, that i was unsure, if people always set a size, so it would be some kind of fallback to have some size value. But the exception would at least notify the user. its up to you :)

Reply:
Yes please. The `update` should only be able to update existing values as the name implies. Just throw if the value to update does not exist since something went wrong then.